### ISSUE TITLE
sanity check the new ( includes neutral phase ) stats output for 2D mld

### ISSUE THREAD

**Issue by Thomas-Moore-Creative** [Thomas-Moore-Creative/Climatology-generator-demo#8]
sanity checking `/g/data/es60/users/thomas_moore/clim_demo_results/daily/draft_delivery/BRAN2020_clim_demo_mld.nc`

**Comment by Thomas-Moore-Creative** [comment]
# file size = 135G !!!??!!

why?

previous version was around 6.6GB and inspection of current `mld` object shows only 7.25GB of float32 data?

```
mld_stats_dataset.nbytes/1e9

7.257640896

```

the change made to the workflow was an attempt to **turn on** compression:

```
# netcdf4 settings & chunking
compression_opts = {
    'zlib': True,      # Enable compression
    'complevel': 1,    # Set a moderate compression level
    'chunksizes': None,  # adjust based on your data's dimensions
}

# Apply the same compression options to all variables for simplicity
encoding = {var: compression_opts for var in result_ds.data_vars}

# Save to a NetCDF file with compression
result_ds.to_netcdf(results_path+results_file,engine='netcdf4',encoding=encoding)
```
Bu obviously adding these setting makes things worse, much worse.  Maybe there is already default compression?

Given how long the processing takes it might make more sense to:

- write `netcdf` output in default manner
- revisit compression along with other final tweaks to `netcdf` output

Also: do we really know if this doesn't change the chunking? 

`'chunksizes': None,  # adjust based on your data's dimensions`

**Comment by Thomas-Moore-Creative** [comment]
given we have a 135GB (!!!?) `BRAN2020_clim_demo_mld.nc` file can we deflate it? Write post-processing code for draft `netcdf` output #9

**Comment by Thomas-Moore-Creative** [comment]
compression function sorted by progress in #9

**Comment by Thomas-Moore-Creative** [comment]
plotting all 28 variables for July
![plot](https://github.com/Thomas-Moore-Creative/Climatology-generator-demo/assets/36463906/7ad45427-b4b0-466b-9a59-020f92042642)


```
import matplotlib.pyplot as plt

# Get the variable names from the dataset
variable_names = toy_data.data_vars.keys()

# Calculate the number of rows and columns for subplots
num_vars = len(variable_names)
num_rows = int(np.sqrt(num_vars))
num_cols = int(np.ceil(num_vars / num_rows))

# Create a figure and axes for subplots
fig, axes = plt.subplots(num_rows, num_cols, figsize=(20, 10), sharex=True, sharey=True)

# Flatten the axes array if necessary
if num_vars == 1:
    axes = np.array([axes])

# Plot each variable on a separate subplot
for i, var_name in enumerate(variable_names):
    row = i // num_cols
    col = i % num_cols
    toy_data[var_name].mean('month').plot(ax=axes[row, col], label=var_name,add_labels=False)

    # Set subplot title
    axes[row, col].set_title('\n'.join(textwrap.wrap(var_name, 13)))
    

    # Remove coordinate labels
    axes[row, col].set_xticklabels([])
    axes[row, col].set_yticklabels([])

# Remove empty subplots
if num_vars < num_rows * num_cols:
    for i in range(num_vars, num_rows * num_cols):
        fig.delaxes(axes.flatten()[i])

# Adjust spacing between subplots
fig.tight_layout()

# Show the plot
plt.show()

```

**Comment by Thomas-Moore-Creative** [comment]
# July sections - equator and 160W - neutral & el niño &  La Niña
![BRAN2020_equator_mld_july](https://github.com/Thomas-Moore-Creative/Climatology-generator-demo/assets/36463906/4a908d88-5660-4e72-860c-93b7a0baa946)

![BRAN2020_equator_el_nino_mld_july](https://github.com/Thomas-Moore-Creative/Climatology-generator-demo/assets/36463906/209169ba-2ddc-426f-b6de-2c03f293a362)

![BRAN2020_equator_la_nina_mld_july](https://github.com/Thomas-Moore-Creative/Climatology-generator-demo/assets/36463906/72fb3155-0e7b-4d21-be30-22d9bfa545e8)

![BRAN2020_160W_mld_july](https://github.com/Thomas-Moore-Creative/Climatology-generator-demo/assets/36463906/9f16939e-d86c-4d01-bafe-137adbadfc09)

**Comment by Thomas-Moore-Creative** [comment]
@matt-csiro 

The `mld` output sanity checks are looking good so far.

The new code with both a fix for the quantile bug #7 and adding the `neutral` phase is producing 2D `netcdf` in about 1.5 hours on the `normal-bw` ARE queue. With `xarray` defaults the 7.25GB of `float32` ends up at 6.8GB.  But simple compression code #9 ( level 4 or 5 ) reduced that to about 3.3GB in 3 minutes.

`eta_t` has also been run but not sanity checked yet.  Now to scale things up on Gadi Cascade Lake nodes for the 3D runs.

NB: for reasons I won't explain the current draft 2D output files are one directory deeper at:
`/g/data/es60/users/thomas_moore/clim_demo_results/daily/draft_delivery/safe/`
